maraboupy/MarabouNetworkONNX.py

```diff
@@ -761,13 +768,149 @@ def matMulEquations(self, node, makeEquations):
+        for attr in node.attribute:
+            if attr.name == "axis":
+                axis = get_attribute_value(attr)
+
+        # Set maps of shape and var
+        inputVars = list([self.varMap[input] for input in node.input])
+        outputVars = np.concatenate(inputVars, axis)
+        self.shapeMap[nodeName] = outputVars.shape
+        self.varMap[nodeName] = outputVars 
+
+    def splitEquations(self, node, nodeName, makeEquations):
+        """Function to generate equations corresponding to split
+
+        Args:
+            node (node): ONNX node representing the Split operation
+            nodeName (str): Name of target node
+            makeEquations (bool): True if we need to create new variables and write Marabou equations
+
+        :meta private:
+        """
+        # Get attributes
+        axis = None
+        split = None
+        for attr in node.attribute:
+            if attr.name == "axis":
+                axis = get_attribute_value(attr)
+            if attr.name == "split":
+                split = get_attribute_value(attr)
+        
+        inputName = node.input[0]
+        inputVars = torch.from_numpy(self.varMap[inputName]) # rely on torch since split opereation of numpy behaves differently from that of onnx
+        inputVars = inputVars.split(split, axis) # tuple
+
+        assert len(inputVars) == len(node.output)
+
+        # Set a shape of target output
+        for i in range(len(node.output)):
+            if node.output[i] == nodeName:
+                self.shapeMap[node.output[i]] = inputVars[i].numpy().shape
+                break
+
+        if not makeEquations:
+            return
+
+        # Get variables and add quations
+        for i in range(len(node.output)):
+            if node.output[i] == nodeName:
+                reshapedInputVars = inputVars[i].reshape(-1)
+                outputVars = self.makeNewVariables(node.output[i]).reshape(-1)
+                for j in range(len(reshapedInputVars)):
+                    # Add equation
+                    e = MarabouUtils.Equation()
+                    e.addAddend(-1, outputVars[j])
+                    e.addAddend(1, reshapedInputVars[j])
+                    e.setScalar(0)
+                    self.addEquation(e)
+                break
+
+    def resizeEquations(self, node, makeEquations):
+        """Function to generate equations corresponding to resize
+
+        Args:
+            node (node): ONNX node representing the Resize operation
+            makeEquations (bool): True if we need to create new variables and write Marabou equations
+
+        :meta private:
+        """
+        nodeName = node.output[0]
+        inputName = node.input[0]
+
+        # Check numper of dimension of input
+        inputVars = self.varMap[inputName]
+        inputShape = inputVars.shape
+        if inputVars.ndim != 4:
+            raise NotImplementedError("Marabou only supports resize operator for very specific upsample case used in YOLO now.")
```

Comment:
Could you add a comment here describing the currently unsupported functionality of resize?

Reply:
added